### ISSUE TITLE
matplotlibrc file in working directory is ignored when running from Spyder

### ISSUE THREAD

**Issue by schtandard** [spyder-ide/spyder#21854]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder
    * [x] Reset preferences with ``spyder --reset``
    * [x] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [x] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

A `matplotlibrc` file in the working directory is ignored when running a script inside Spyder. It works as expected when running from the console (using either `python` or `ipython`).

### What steps reproduce the problem?

1. Create the file `rcfile.py` with the following content:
    ```python
    import os
    from matplotlib import pyplot as plt
    
    assert 'matplotlibrc' in os.listdir()
    
    # With the following line, the style sheet is used.
    # plt.style.use('matplotlibrc')
    
    fig, ax = plt.subplots()
    fig.savefig('test.png')
    ```
2. In the same directory, create the file `matplotlibrc` with the following content:
    ```
    axes.edgecolor: red
    ```
3. Open `rcfile.py` in Spyder and run it. It outputs this image:

    ![test](https://github.com/spyder-ide/spyder/assets/23075534/9a978ca2-0ca1-4b44-a780-af351dba454a)

4. Run `python rcfile.py` from the command line. It outputs this image:

    ![test](https://github.com/spyder-ide/spyder/assets/23075534/94866115-f14f-49bf-8850-5b0ffa113599)

I could not find any documentation of this behavior. The resolution of the two images is also different, presumably due to the setting _Preferences -> IPython console -> Graphics -> Inline backend -> Resolution_, though it is not obvious from the setting that it changes `rcParams` rather than just controlling how the `png` displayed in Spyder is created.

Explicitly loading the `matplotlibrc` configuration does work.

## Versions

* Spyder version: 5.4.3
* Python version: 3.11.8
* Qt version: 5.15.2
* PyQt version: 5.15.10
* Operating System name/version: Windows 11

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0                             :  1.4.0 (OK)
chardet >=2.0.0                                  :  4.0.0 (OK)
cloudpickle >=0.5.0                              :  2.2.1 (OK)
cookiecutter >=1.6.0                             :  2.5.0 (OK)
diff_match_patch >=20181111                      :  20200713 (OK)
intervaltree >=3.0.2                             :  3.1.0 (OK)
IPython >=7.31.1,<9.0.0,!=8.8.0,!=8.9.0,!=8.10.0 :  8.20.0 (OK)
jedi >=0.17.2,<0.19.0                            :  0.18.1 (OK)
jellyfish >=0.7                                  :  1.0.1 (OK)
jsonschema >=3.2.0                               :  4.19.2 (OK)
keyring >=17.0.0                                 :  23.13.1 (OK)
nbconvert >=4.0                                  :  7.10.0 (OK)
numpydoc >=0.6.0                                 :  1.5.0 (OK)
paramiko >=2.4.0                                 :  2.8.1 (OK)
parso >=0.7.0,<0.9.0                             :  0.8.3 (OK)
pexpect >=4.4.0                                  :  4.8.0 (OK)
pickleshare >=0.4                                :  0.7.5 (OK)
psutil >=5.3                                     :  5.9.0 (OK)
pygments >=2.0                                   :  2.15.1 (OK)
pylint >=2.5.0,<3.0                              :  2.16.2 (OK)
pylint_venv >=2.1.1                              :  2.3.0 (OK)
pyls_spyder >=0.4.0                              :  0.4.0 (OK)
pylsp >=1.7.2,<1.8.0                             :  1.7.2 (OK)
pylsp_black >=1.2.0                              :  2.0.0 (OK)
qdarkstyle >=3.0.2,<3.2.0                        :  3.0.2 (OK)
qstylizer >=0.2.2                                :  0.2.2 (OK)
qtawesome >=1.2.1                                :  1.2.2 (OK)
qtconsole >=5.4.2,<5.5.0                         :  5.4.2 (OK)
qtpy >=2.1.0                                     :  2.4.1 (OK)
rtree >=0.9.7                                    :  1.0.1 (OK)
setuptools >=49.6.0                              :  68.2.2 (OK)
sphinx >=0.6.6                                   :  5.0.2 (OK)
spyder_kernels >=2.4.3,<2.5.0                    :  2.4.4 (OK)
textdistance >=4.2.0                             :  4.2.1 (OK)
three_merge >=0.1.1                              :  0.1.1 (OK)
watchdog >=0.10.3                                :  2.1.6 (OK)
zmq >=22.1.0                                     :  25.1.2 (OK)

# Optional:
cython >=0.21                                    :  None (NOK)
matplotlib >=3.0.0                               :  3.8.0 (OK)
numpy >=1.7                                      :  1.26.4 (OK)
pandas >=1.1.1                                   :  None (NOK)
scipy >=0.17.0                                   :  None (NOK)
sympy >=0.7.3                                    :  None (NOK)

```

**Comment by dalthviz** [comment]
Hi @schtandard thank you for the feedback! I did a quick test and I was able to reproduce this :+1: Also, maybe a way to workaround this for the moment would be to set the Spyder working directory to the directory where your files are and then restart the Spyder IPython console kernel (using the `Restart kernel` action). Let us know if the workaround helps!

**Comment by schtandard** [comment]
Do you mean using _Preferences -> Run -> Working directory settings -> The following direcotry_? That doesn't seem to work either: Now the `assert` fails, as Spyder seems to run the script from my user directory (`C:\Users\wilde`), no matter which directory I set in the preferences. Did this work on your end?

However, even if this worked, it wouldn't be an option for me, as I usually work on several files located in different directories at the same time and need each one to be executed from their own working directory, so setting a fixed one is not an option. For the time being, I will just load the `matplotlibrc` file manually.

**Comment by dalthviz** [comment]
I meant to say to use the working directory toolbar, sorry for the confusion there 😅 :

![image](https://github.com/spyder-ide/spyder/assets/16781833/f77448c3-d747-48d0-a5e3-ba4d4f79475b)

What works for me as another workaround is to be sure that the correct working directory is set in the toolbar plus restarting the console kernel (which maybe means that somehow the `matplolibrc` load is done just once per kernel run 🤔, what do you think about that @ccordoba12 ?):

![matplotlibrc](https://github.com/spyder-ide/spyder/assets/16781833/2a34a097-0174-40db-86ec-8e248e2dc15e)

And indeed, seems way easier to just force the `matplolibrc` load programatically as a workaround :+1:

**Comment by ccordoba12** [comment]
> which maybe means that somehow the matplolibrc load is done just once per kernel run 🤔, what do you think about that @ccordoba12 ?

Yeah, I'd say that's the right assessment @dalthviz because most probably Matplotlib can be configured through `matplotlibrc` only once, when it's imported.

However, I'm not sure how to tackle this issue. Perhaps we could detect if there's a `matplotlibrc` in the current working directory before executing a file and, if that's the case, display a message in the console saying something like: 

> We detected that you have a `matplotlibrc` file in your working directory. Please be aware that any changes or additions to it require a kernel restart to take effect.
